app/models/user.rb

```diff
@@ -11,22 +11,32 @@ class User < ApplicationRecord
   validates :password, presence: true,
     length: {minimum: Settings.password_min_length}, allow_nil: true
 
+  before_create :create_activation_digest
   before_save ->{email.downcase!}
 
   def remember
     self.remember_token = User.new_token
     update_attributes remember_digest: User.digest(remember_token)
   end
 
-  def authenticated? remember_token
-    return false if remember_digest.blank?
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def authenticated? attribute, token
+    digest = send("#{attribute}_digest")
+    return false if digest.blank?
```

Comment:
`if digest.blank?` -> `unless digest` thì sao chú

Reply:
dạ em sợ digest nó toàn là kí tự trống nên mới để như vậy ạ. Nhưng có vẻ trong trường hợp này digest là chuỗi hash nên không trống được. Em sẽ sửa ạ.